diagnostics-extension/src/services/events.ts

```diff
@@ -0,0 +1,119 @@
+import * as fake_events from './fake_events.data';
+import { environment } from '../environments/environment';
+import { generateEventsSuccessResponse, generateErrorResponse } from '../utils';
+
+/**
+ * Abstract class reprensenting the interface of
+ * service to handle events
+ */
+export abstract class EventsService {
+  abstract registerEventHandlers(): Promise<Response>;
+  abstract registerPort(port: chrome.runtime.Port): void;
+  abstract isEventSupported(eventType: EventCategory): Promise<EventSupportStatusInfo>;
+  abstract startCapturingEvents(eventType: EventCategory): Promise<void>;
+  abstract stopCapturingEvents(eventType: EventCategory): Promise<void>;
+}
+
+/**
+ * Implementation of EventsService
+ * @extends EventsService
+ */
+export class EventsServiceImpl extends EventsService {
+  private port!: chrome.runtime.Port;
+
+  notifyPort = (type: EventCategory, event) => {
+    this.port.postMessage({
+      type: type,
+      info: event
+    })
+    return;
+  }
+
+  registerPort = (port: chrome.runtime.Port): void => {
+    this.port = port;
+  }
+
+  registerEventHandlers = async (): Promise<Response> => {
+    try {
+      (chrome as any).os.events.onAudioJackEvent.addListener(this.notifyPort.bind(null, EventCategory.audio_jack));
+      (chrome as any).os.events.onExternalDisplayEvent.addListener(this.notifyPort.bind(null, EventCategory.external_display));
+      (chrome as any).os.events.onKeyboardDiagnosticEvent.addListener(this.notifyPort.bind(null, EventCategory.keyboard_diagnostic));
+      (chrome as any).os.events.onLidEvent.addListener(this.notifyPort.bind(null, EventCategory.lid));
+      (chrome as any).os.events.onSdCardEvent.addListener(this.notifyPort.bind(null, EventCategory.sd_card));
+      (chrome as any).os.events.onStylusConnectedEvent.addListener(this.notifyPort.bind(null, EventCategory.stylus_connected));
+      (chrome as any).os.events.onStylusGarageEvent.addListener(this.notifyPort.bind(null, EventCategory.stylus_garage));
+      (chrome as any).os.events.onStylusTouchEvent.addListener(this.notifyPort.bind(null, EventCategory.stylus_touch));
+      (chrome as any).os.events.onTouchpadButtonEvent.addListener(this.notifyPort.bind(null, EventCategory.touchpad_button));
+      (chrome as any).os.events.onTouchpadConnectedEvent.addListener(this.notifyPort.bind(null, EventCategory.touchpad_connected));
+      (chrome as any).os.events.onTouchpadTouchEvent.addListener(this.notifyPort.bind(null, EventCategory.touchpad_touch));
+      (chrome as any).os.events.onTouchscreenConnectedEvent.addListener(this.notifyPort.bind(null, EventCategory.touchscreen_connected));
+      (chrome as any).os.events.onTouchscreenTouchEvent.addListener(this.notifyPort.bind(null, EventCategory.touchscreen_touch));
+      (chrome as any).os.events.onUsbEvent.addListener(this.notifyPort.bind(null, EventCategory.usb));
+      (chrome as any).os.events.onPowerEvent.addListener(this.notifyPort.bind(null, EventCategory.power));
```

Comment:
Can we use an array of struct here?

Something like
```
[
{
  type: EventCategory.audio_jack,
  func: (chrome as any).os.events.onAudioJackEvent
},
...
]
```

And we can iterate over the array to add listener. 

This way we can easily add new event types in the future

Reply:
Done.